### ISSUE TITLE
process of making heatmaps

### ISSUE THREAD

**Issue by Michael3444** [michalfaber/keras_Realtime_Multi-Person_Pose_Estimation#75]
Hey bro,thanks for your great work!, I have recently read your codes and learned a lot. but I also got some confusions about the process of makig heatmaps. In the original paper:''The groundtruthconfidence map to be predicted by the network is an aggregation of the individual confidence maps via a max operator'',but in the original caffe  cpm_data_transformer.cpp and your label_maps.py, you both just add them directly,I don't know if it has any impact to the final results.
Then,when i tried to train mscoco-2017 using your newest codes(tensorpack IO),It didn't trained well ,the total loss is about 2000 and stageN_l2_loss is about 240,I calculate the 184 lines:train_samples = df.size() in train_pose.py  it's 11970 but the total items is 119707,it changed with the batchsize, train_samples=total_items/batchsize,Hope your reply !

**Comment by Michael3444** [comment]
I didn't know if somewhere went wrong or other some mistakes

**Comment by michalfaber** [comment]
Hi @Michael3444 
Great thanks for pointing out a bug. There should be 119707 samples, not 11970. I’ve just fixed that.
Yes, this implementation is slightly different from the original paper and is heavily based on the cmu work and theirs cpm_data_transformer.cpp Right now I am replacing my old deep learning rig so I hope next month I will start doing more experiments. Thanks

**Comment by Michael3444** [comment]
@michalfaber  Hi bro,Did you train the models using your newest codes? I found that the L2_loss is always three times of the L1_loss ,during my third epoch,L1_loss was about 80 and L2_loss was about 240:

> 6271/11970 [==============>...............] - ETA: 5611s - loss: 1951.7319 - weight_stage1_L1_loss: 88.3196 - weight_stage1_L2_loss: 246.0139 - weight_stage2_L1_loss: 80.5156 - weight_stage2_L2_loss: 243.0767 - weight_stage3_L1_loss: 79.9292 - weight_stage3_L2_loss: 242.8260 - weight_stage4_L1_loss: 79.7837 - weight_stage4_L2_loss: 242.7326 - weight_stage5_L1_loss: 79.6420 - weight_stage5_L2_loss: 242.7095 - weight_stage6_L1_loss: 79.2863 - weight_stage6_L2_loss: 242.6080 
I could't reimplement the original paper ,I had checked your codes but  couldn't find somewhere that went wrong ,it bothered me a lot! 
Besides ,about the problem of stopIteration at the end of each epoch , the class of RepeatedData could help fix that 

```
def batch_dataflow(df,batch_size):
    df = RepeatedData(df,-1)
    df = BatchData(df, batch_size, use_list=False)
    df = MapData(df, lambda x: (
        [x[0], x[1], x[2]],
        [x[3], x[4], x[3], x[4], x[3], x[4], x[3], x[4], x[3], x[4], x[3], x[4]])
    )
    #df.reset_state()
    return df
```
 Hope your reply!
